pure.zsh

```diff
@@ -114,13 +132,15 @@ prompt_pure_preprompt_render() {
 		# only redraw if preprompt has changed
 		[[ "${prompt_pure_last_preprompt}" != "${preprompt}" ]] || return
 
-		# calculate length of preprompt for redraw purposes
-		local preprompt_length=$(prompt_pure_string_length $preprompt)
-		local lines=$(( ($preprompt_length - 1) / $COLUMNS + 1 ))
+		# calculate preprompt length and store it in prompt_pure_preprompt_length
+		prompt_pure_string_length_to_var "${preprompt}" "prompt_pure_preprompt_length"
```

Comment:
Oh, that's cool. I've read that about `-g` before, but this kind of use case didn't register for me!